finality-aleph/src/sync/service.rs

```diff
@@ -69,12 +81,20 @@ impl<
         finalizer: F,
         period: SessionPeriod,
         additional_justifications_from_user: mpsc::UnboundedReceiver<J::Unverified>,
-    ) -> Result<(Self, impl JustificationSubmissions<J> + Clone), HandlerError<J, CS, V, F>> {
+    ) -> Result<
+        (
+            Self,
+            impl JustificationSubmissions<J> + Clone,
+            impl RequestBlocks<<<J as Justification>::Header as Header>::Identifier>,
```

Comment:
Hmm, what is the benefit of returning impl here over specific type?

Reply:
just duplicating what was already here, but I guess this communicates the intention much better than saying "it's a channel", and there's only, tiny, compilation time cost, right?

@timorl - would you explain more pls?

Reply:
The compiler ensures that external code will not depend on any other properties of the returned type, which makes replacing it later with something else much simpler and generally ensures that we are doing what we expect. The smaller the exposed API, the better.